### PR TITLE
Add scope-based permission checks to collaborator listing

### PR DESCRIPTION
**Pull request by AustinSMueller**
Enhanced the listPotentialCollaborators method to enforce permission checks based on user scope (none, role, own, study, DC) using _getUserScope. Updated and expanded tests to cover all scope scenarios and edge cases, ensuring correct access control and error handling.

### CODE REVIEW

File: test/services/submission.listPotentialCollaborators.test.js [hunk 1]
```diff
@@ -197,6 +206,27 @@ describe('Submission.listPotentialCollaborators', () => {
...
+        // Final cleanup
+        jest.clearAllMocks();
+        jest.restoreAllMocks();
+        jest.clearAllTimers();
+        
+        // Clean up any remaining handles
+        if (global.gc) {
+            global.gc();
+        }
+    });
```
[the remark this answers is not in the document]
Comment (AustinSMueller) [reply test/services/submission.listPotentialCollaborators.test.js]: cleaned up this teardown code

File: test/services/submission.listPotentialCollaborators.test.js [hunk 2]
```diff
@@ -197,6 +206,27 @@ describe('Submission.listPotentialCollaborators', () => {
...
     });
 
+    afterEach(() => {
+        // Clean up all mocks
+        jest.clearAllMocks();
+        jest.restoreAllMocks();
+        
+        // Clear any timers
+        jest.clearAllTimers();
+    });
```
[the remark this answers is not in the document]
Comment (AustinSMueller) [reply test/services/submission.listPotentialCollaborators.test.js]: cleaned up this teardown code
